src/MoBi.Presentation/Presenter/ChartPresenter.cs

```diff
@@ -190,76 +193,56 @@ protected virtual void OnDragDrop(object sender, DragEventArgs e)
          {
             var historicalResultsNodes = e.Data<IList<ITreeNode>>().OfType<HistoricalResultsNode>();
-            historicalResultsNodes.Each(n => addHistoricalResults(n.Tag));
+            addHistoricalResults(historicalResultsNodes.Select(result => result.Tag).ToList());
          }
          else
          {
             var droppedObservedData = _observedDataDragDropBinder.DroppedObservedDataFrom(e);
-            droppedObservedData.Each(addObservedData);
+            addObservedData(droppedObservedData.ToList());
          }
       }
 
       protected abstract bool CanDropSimulation { get; }
 
       protected abstract void MarkChartOwnerAsChanged();
 
-      private void addHistoricalResults(DataRepository repository)
+      private void addHistoricalResults(IReadOnlyList<DataRepository> repositories)
       {
-         addDataRepositorySimulation(repository);
-         _editorPresenter.AddDataRepository(repository);
-         repository.SetPersistable(persistable: true);
+         addDataRepositoriesToDataRepositoryCache(repositories);
+         _editorPresenter.AddDataRepositories(repositories);
+         repositories.Each(repository => repository.SetPersistable(persistable: true));
       }
 
-      private void addObservedData(DataRepository repository)
+      private void addObservedData(IReadOnlyList<DataRepository> repositories)
       {
-         _editorPresenter.AddDataRepository(repository);
-         repository.Where(c => c.DataInfo.Origin == ColumnOrigins.Observation)
-            .Each(c => _editorPresenter.AddCurveForColumn(c.Id));
+         _editorPresenter.AddDataRepositories(repositories);
+         repositories.SelectMany(x => x.ObservationColumns()).Each(observationColumn => _editorPresenter.AddCurveForColumn(observationColumn));
+         _chartPresenterContext.Refresh();
       }
 
       private void addObservedDataIfNeeded(IEnumerable<DataRepository> dataRepositories)
       {
          //curves are already selected. no need to add default selection
          if (Chart.Curves.Any()) return;
-         dataRepositories.Each(addObservedData);
+         addObservedData(dataRepositories.ToList());
       }
 
-      public ICurveChart Chart
-      {
-         get { return _displayPresenter.DataSource; }
-         protected set
-         {
-            if (Chart != null)
-               Chart.PropertyChanged -= onPropertyChanged;
-
-            _editorPresenter.DataSource = value;
-            _displayPresenter.DataSource = value;
-
-            if (Chart != null)
-               Chart.PropertyChanged += onPropertyChanged;
-         }
-      }
-
-      private void onPropertyChanged(object sender, PropertyChangedEventArgs e)
+      protected override void ChartChanged()
       {
+         base.ChartChanged();
          MarkChartOwnerAsChanged();
-         if (e.PropertyName.Equals(_nameProperty))
-         {
-            _context.PublishEvent(new RenamedEvent(Chart));
```

Comment:
Not for the chart. I think this was left when the tab caption was being updated with chart name. Probably some time after that the chart was loaded only when the tab was clicked and this stopped working.

When I stepped through the handlers of RenamedEvent, it's not used for chart.